pkg/webhook/cert/certclient/certclient.go

```diff
@@ -158,58 +158,64 @@ var genericFn = func(current, desired *client.Object) error {
 // createOrReplaceHelper creates the object if it doesn't exist;
 // otherwise, it will replace it.
 // When replacing, fn  should know how to preserve existing fields in the object GET from the APIServer.
-func createOrReplaceHelper(c client.Client, obj client.Object, fn mutateFn) error {
+func createOrReplaceHelper(ctx context.Context, c client.Client, obj client.Object, fn mutateFn) error {
 	if obj == nil {
 		return nil
 	}
-	err := c.Create(context.Background(), obj)
-	if apierrors.IsAlreadyExists(err) {
-		// TODO: retry mutiple times with backoff if necessary.
-		// this cast is not safe per-say but is added to get around the transition from runtime.Object to client.Object
-		existing, ok := obj.DeepCopyObject().(client.Object)
-		if !ok {
-			return fmt.Errorf("unable to cast deep copy to client.Object")
-		}
-		objectKey := client.ObjectKeyFromObject(obj)
-		err = c.Get(context.Background(), objectKey, existing)
-		if err != nil {
-			return err
-		}
-		err = fn(&existing, &obj)
-		if err != nil {
-			return err
+
+	objectKey := client.ObjectKeyFromObject(obj)
+
+	// this cast is not safe per-se but is added to get around the transition from runtime.Object to client.Object
+	existing, ok := obj.DeepCopyObject().(client.Object)
+	if !ok {
+		return fmt.Errorf("unable to cast deep copy %T to client.Object", obj)
+	}
+
+	// NOTE: create *before* get does not always work for services.
```

Comment:
Good to know this! It sounds to apply to all (wondering why the ctrl runtime does not handle it)